Using Documenter's new options

pbastide opened the pull request:
This is a rewrite of the documentation to run Julia example code with `Documenter` instead of `Weave`.

See `docs/readme.md` for details.

For the rewriting to be complete, all the files `docs/src/man/src/*.jmd` must be translated into `docs/src/man/*.md`. When such a file is translated, the associated line in `docs/mkdocs/yml` can be un-commented.

cecileane commented:
@pbastide could you please check?

For `expectedCFs`, we needed to use Gadfly to plot the concordance factors. For another file too (score versus hmax). I chose to avoid Gadfly: it was easy enough to make a simple plot for the score vs. hmax.  But not so to plot concordance factors. I wanted to try to use ggplot. RCall makes it *super easy*, this is such a pleasure. So I wanted to share. But the downside is that the documentation now needs `ggplot2` (and the dependency to `ggsave` to an svg file: `svglite`).

I added a line to `.travis.yml` to install `ggplot2`. But I could not test to see if it works. Could you possibly test that easily? Or should we just merge into the master branch and see?

pbastide commented:
Thank you @cecileane for your advances on the doc !

From the last `Travis` build above, it seems that the installation of `ggplot2` worked (it took around 6 minutes, see [line 1471](https://travis-ci.org/crsl4/PhyloNetworks.jl/jobs/399306521#L1471)).

Just to check, I merged back into my sandbox branch. It does indeed seem to work ! See the deployed test doc [here](http://pbastide.github.io/PhyloNetworks.jl/latest/man/expectedCFs/#Extract-Expected-CFs-1).

cecileane commented:
Very cool thanks for checking!
- 6 minutes is a very long time, given the time limits on travis. I wonder how much time we would save if we installed `ggplot2` *without* dependencies (by this I mean: `dep=FALSE`), and separately installed `svglite`, which we need to create svg plots.
- Also, I thought I converted all files. you mentioned one missing: which one did you have in mind? I'm happy to do it today.

pbastide commented:
* Yes, maybe it's best to install only what we need. I saw you just gave it a try, I can pull back to the sandbox branch again anytime if necessary.
* My bad ! I saw that `docs/src/man/src` still had a `.jmd`, but this is actually just a showcase for `Weave`, so not part of the doc. Sorry for the confusion.

pbastide commented:
* Indeed, it's a local file. Rookie mistake !
* It seems that removing the dependencies did not work above, see e.g. [lines 1499](https://travis-ci.org/crsl4/PhyloNetworks.jl/jobs/399596414#L1499) and [1698](https://travis-ci.org/crsl4/PhyloNetworks.jl/jobs/399596414#L1698). `ggplot2` could not be installed.

I'm not sure I understand your second solution: you want to do and show a base plot, while displaying a ggplot command ? Might that not be confusing to the reader ? In that case, maybe it would be best to show the base plot command. And, if we don't want to install `ggplot2`, we can show the base plot, and then add as a remark that the plot could be obtained with `ggplot2`, and show the not-executed commands ?

cecileane commented:
what do you think of this option? I used basic plot commands, but picked the colors and transparency to roughly match what we get with ggplot2. I also changed the ggplot2 command to the classic theme, to match what we get with basic plots roughly. If you think that we should show both sets of commands, then sure. I just wanted to keep things simpler for the reader ("less is more"). But in any case, 6 travis minutes out of a max of 30 (?) is too much, I think.

pbastide commented:
It's a bit of a hack, but I agree this version should be less confusing than showing the base R commands. Let's go with it !

I pulled the doc to check: everything looks fine, see [here](http://pbastide.github.io/PhyloNetworks.jl/latest/man/expectedCFs/#Extract-Expected-CFs-1).
Just, on the second figure, the line is asked "black" on the ggplot, so I changed it in the associated base R. Also, the legend was not visible, so I moved it a bit.
I tried to also move the legend in the ggplot, but I could not manage to do it. I get the following error:
```julia
julia> ggplot(df_long, aes(x=:obsCF, y=:expCF, color=:has_A)) + theme_classic() +
                 geom_segment(x=0,y=0,xend=1,yend=1, color="black", size=0.3) +
                 geom_point(alpha=0.5, position=position_jitter(width=0.005, height=0.005)) +
                 ylab("quartet CF expected from network") + xlab("quartet CF observed in gene trees") + coord_equal(ratio=1) +
                 theme(legend.position=[0.7,0.3]);

ERROR: syntax: keyword argument is not a symbol: "legend.position"
```
I think there is a problem with the '.' in 'legend.position', but I could not go around it. It's not a big deal, but slightly annoying. Do you know of a simple way to do this ?

Otherwise, everything seems fine !

cecileane commented:
yes, the '.' in R cause issues in Julia. This very cool [blog post](http://avt.im/blog/2018/03/23/R-packages-ggplot-in-julia) shows ways to deal with this. I think it's just fine if the ggplot command places the legend in a different place than what's shown on the website. So we can go ahead and merge.

cecileane commented:
I don't see the plots unfortunately. see [latest trait doc](http://crsl4.github.io/PhyloNetworks.jl/latest/man/trait_tree/) for instance...

pbastide commented:
Could this be just an issue related to the merging ?

The doc page you point to does not have the latest changes: for instance, it does not have the warning we added about model orders in function `ftest` (that should be in [this section](http://crsl4.github.io/PhyloNetworks.jl/latest/man/trait_tree/#Fit-of-a-Shifted-BM-1)).

On the [gh-pages](https://github.com/crsl4/PhyloNetworks.jl/tree/gh-pages/latest) branch, the figures are based on the latest merge commit (6204bc4), but the man pages on the one before (e29fc38). Maybe the issue could go away at the next commit on master ? (I'm optimistic...)

pbastide commented:
Actually, that might be a timing issue: the build of 6204bc4 was finished at 5:16pm GMT+2, while the build of e29fc38 was finished at 5:20 GMT+2.

It is coherent with the build times on Travis: e29fc38 took 43 min 26 sec; while 6204bc4 took only 37 min 5 sec...

So what might have happen is that the doc built on the older commit got deployed *after* the one built on the latest commit, and hence replaced it ! Would it make sense ?

cecileane commented:
Yes! You are an awesome detective Paul. I think you are right. Let me do another commit. I am working on adding new content to the documentation now, so I will have plenty of new opportunities for commits.
